src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/JsonSourceGenerator.cs

```diff
@@ -24,33 +25,44 @@ public class JsonSerializerSourceGenerator : ISourceGenerator
         public void Execute(SourceGeneratorContext context)
         {
             JsonSerializableSyntaxReceiver receiver = (JsonSerializableSyntaxReceiver)context.SyntaxReceiver;
-
             MetadataLoadContext metadataLoadContext = new MetadataLoadContext(context.Compilation);
 
-            INamedTypeSymbol namedTypeSymbol;
-            ITypeSymbol typeSymbol;
-            IdentifierNameSyntax identifierNameNode;
-            SemanticModel semanticModel;
-            Type convertedType;
-            TypeDeclarationSyntax typeDeclarationNode;
+            // Filter classes and structs with JsonSerializable attribute semantically.
+            INamedTypeSymbol jsonSerializableAttributeSymbol = context.Compilation.GetTypeByMetadataName("System.Text.Json.Serialization.JsonSerializableAttribute");
```

Comment:
Could this be `null`/`default` and should we exit early if so?

Reply:
If not found GetTypeByMetadataName returns null. Updated it to return if not found.
Also, created an issue to find a way to add messages to user to give more verbose error messages to handle cases like these. https://github.com/dotnet/runtimelab/issues/44